client/src/dialogs/documentswindow.cpp

```diff
@@ -265,11 +265,7 @@ void DocumentsWindow::deleteDocument()
     if (answer == QMessageBox::No)
         return;
 
-    auto it = std::find_if(mDocuments.constBegin(), mDocuments.constEnd(), [&documentId](const SugarDocument &document) { return document.id() == documentId; });
-    if (it != mDocuments.constEnd()) {
-        mDocuments.remove(std::distance(mDocuments.constBegin(), it));
-    }
-
+    // remove only from the view(mDocumentWidgets) at this moment, the data(mDocuments) will be removed at the 'DocumentsWindow::saveChanges'
```

Comment:
This comment confuses me. `saveChanges()` does not remove from `mDocuments`, it deletes from akonadi (ItemDeleteJob).

Can you adjust the comment to something like this?
    // Remove the document from the view (mDocumentWidgets). Keep in mDocuments so that saveChanges() sees it and triggers deletion.

Reply:
Done